client/src/components/HomePage/Registration.jsx

```diff
@@ -29,13 +60,17 @@ class Registration extends Component {
     };
 
     if (
-      !emailValidation(this.state.email)
-      || !this.state.password.length
-      || this.state.password !== this.state.repeat_password
+      !this.state.password.length
     ) {
-      this.setState({ check_password: "passwords isn't equal" });
+      this.setState({ check_password: "Empty field" });
+      window.Materialize.toast("Invalid input", 3000);
+    }
+    if (!emailValidation(this.state.email)) {
+      this.setState({ check_email: "Please provide a valid email address" });
+    }
+    if (!this.state.nickname.length) {
+      this.setState({ check_nickname: "Empty field" });
       window.Materialize.toast("Invalid input", 3000);
-      return null;
     }
 
```

Comment:
I think better to stop function if you have some errors before sending a request to the server,